src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -1868,16 +1868,69 @@ private static Expression BindParameterFromValue(ParameterInfo parameter, Expres
         var parameterNameConstant = Expression.Constant(parameter.Name);
         var sourceConstant = Expression.Constant(source);
 
+        if (source == "header" && (parameter.ParameterType.IsArray || typeof(IEnumerable<string>).IsAssignableFrom(parameter.ParameterType)))
+        {
+            var stringValuesExpr = Expression.Convert(valueExpression, typeof(StringValues));
+            var toStringArrayMethod = typeof(StringValues).GetMethod(nameof(StringValues.ToArray));
+            if (toStringArrayMethod == null)
+            {
+                throw new InvalidOperationException("The method 'ToArray' could not be found on 'StringValues'.");
+            }
```

Comment:
```suggestion
            var toStringArrayMethod = typeof(StringValues).GetMethod(nameof(StringValues.ToArray))!;
```

The method exists, we're referencing it in `nameof`.